src/plugins/controls/common/types.ts

```diff
@@ -7,8 +7,11 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-export type ControlWidth = 'small' | 'medium' | 'large';
-export type ControlLabelPosition = 'twoLine' | 'oneLine';
+import { CONTROL_LABEL_POSITION_OPTIONS, CONTROL_WIDTH_OPTIONS } from './constants';
+
+export type ControlWidth = (typeof CONTROL_WIDTH_OPTIONS)[keyof typeof CONTROL_WIDTH_OPTIONS];
```

Comment:
Same here, `keyof typeof CONTROL_WIDTH_OPTIONS` is the sam e as `(typeof CONTROL_WIDTH_OPTIONS)[keyof typeof CONTROL_WIDTH_OPTIONS]`

Reply:
See https://github.com/elastic/kibana/pull/193067#discussion_r1785136328